tracing-opentelemetry/src/subscriber.rs

```diff
@@ -1188,6 +1212,50 @@ mod tests {
         assert_eq!(recorded_trace_id, trace_id)
     }
 
+    #[test]
+    fn trace_id_propagates_to_current_context() {
+        let tracer = TestTracer(Arc::new(Mutex::new(None)));
+        let subscriber =
+            tracing_subscriber::registry().with(subscriber().with_tracer(tracer.clone()));
+        let trace_id = otel::TraceId::from(42u128.to_be_bytes());
+        let existing_cx = OtelContext::current_with_span(TestSpan(otel::SpanContext::new(
+            trace_id,
+            otel::SpanId::from(1u64.to_be_bytes()),
+            TraceFlags::default(),
+            false,
+            Default::default(),
+        )));
+
+        tracing::collect::with_default(subscriber, || {
+            let outer_span = tracing::debug_span!("outer");
+            outer_span.set_parent(existing_cx);
+            let _outer_g = outer_span.enter();
+
+            let inner_span = tracing::debug_span!("inner");
+
+            dbg!(OtelContext::current().span().span_context().trace_id());
+            dbg!(OtelContext::current().has_active_span(),);
+            dbg!(outer_span.context().has_active_span(),);
+            dbg!(outer_span.context().span().span_context().trace_id());
+            // let inner_span = tracing::debug_span!("inner");
+            dbg!(tracing::Span::current().id().unwrap());
+            assert_eq!(
+                OtelContext::current().span().span_context().trace_id(),
+                trace_id
+            );
+            let _inner_g = inner_span.enter();
+            assert_eq!(
+                OtelContext::current().span().span_context().trace_id(),
+                trace_id
+            );
```

Comment:
`inner_span` is not holding a parent, this currently evaluates to `None`:
https://github.com/tokio-rs/tracing/blob/eca60a6b4363c69bade0a7d7d4eb0554e2a64267/tracing-opentelemetry/src/subscriber.rs#L591